### PR TITLE
Add `kubeadm upgrade` docs

### PR DESCRIPTION
**Pull request by mhausenblas**
addresses kubernetes/kubernetes.github.io/issues/4689

note this is the first draft, just to get the ball rolling ;)

CC: @luxas 

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4770)

### CODE REVIEW

File: _data/tasks.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
-bigheader: "Tasks"
-abstract: "Step-by-step instructions for performing operations with Kubernetes."
+bigheader:   "Tasks"
```
Comment (chenopis) [inline _data/tasks.yml]: Please revert the addition of the extra spacing.

Comment (mhausenblas) [inline _data/tasks.yml]: Oops, sorry, didn't notice this. It's due to an Atom package that automatically formats YAML files.

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+{% capture prerequisites %}
+You need to have a Kubernetes cluster running version 1.7.x in order to use the process described here. Note that only one minor version upgrade is supported, that is, you can only upgrade from, say 1.7 to 1.8, not from 1.7 to 1.9.
+
+Before proceeding, make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#v180-alpha2) carefully.
+
+The following is out of scope for `kubeadm upgrade`, that is, you need to take care of it yourself:
+
+- No etcd upgrades are performed. You can, for example, use `etcdctl` to take care of this.
+- Any app-level state, for example, a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Can we break this out to a section where you're saying the general: "As a best-practice; you should back up what's important to you" or something like that
We should clarify that this _should not touch any workloads_, just the k8s internal components, but it's always good to be on the safe side

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+---
+approvers:
+- pipejakob
```
Comment (mhausenblas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: he he, a copy and paste thing, yes makes sense …

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+Note that `kubeadm upgrade` is 'eventually idempotent', that is, you can run it over and over again if you find yourself in a bad state and it should be able to recover.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## On the master
+
+1. Upgrade `kubectl` using [curl](/docs/tasks/tools/install-kubectl/#install-kubectl-binary-via-curl). Note: DO NOT use `apt` or `yum` or any other package manager to upgrade it.
```
Comment (mhausenblas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Hmmm, now I'm a bit confused. At least that was the state when we discussed it. Has this changed or did I misunderstand you here?

**Comment by luxas** [comment]
@mhausenblas please ping me when ready so I know

**Comment by mhausenblas** [comment]
@luxas sure thing, scheduled for beginning of next week. Will ping you then …

**Comment by steveperry-53** [comment]
SUGGESTED CHANGES TO THE TEXT

**Opening paragraph**

Remove comma between cluster and currently.

**Before you Begin**

Change "You need to have a Kubernetes cluster" to "You need to have a self-hosted Kubernetes cluster" or "You need to have a kubeadm Kubernetes cluster".

Put the note about one minor version somewhere outside of the **Before you begin** section.

Does upgrading etcd have to be done before you begin? If not, move the note out of the **Before you begin** section.

Make the steps more concise. For example:

* You need to have a self-hosted Kubernetes cluster running version 1.7.x.
* Read the release notes.
* Back up any app-level state. For example ...

Move the note about being eventually idempotent outside of the **Before you begin** section.

**On the master**

Include a link to using curl to install the most recent version of kubeadm.

Spell out Software Defined Network.

Spell out Container Network Interface once.

No need to say, "After kubeadm upgrade you need to". Just say "Manually upgrade your Software Defined Network."

**Recovering from a bad state**

I don't understand what's being said here. Maybe rephrase and blend in the statement about kubeadm upgrade being eventually idempotent.

**Comment by kargakis** [comment]
@mhausenblas any updates here?

**Comment by mhausenblas** [comment]
@kargakis I'm on it

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+See also [upgrading kubeadm clusters from 1.6 to 1.7](/docs/tasks/administer-cluster/kubeadm-upgrade-1-7/) if you're on a 1.6 cluster currently.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a `kubeadm` Kubernetes cluster running version 1.7.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#v180-alpha2) carefully.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: we don't have to pin this to alpha2

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Upgrading your control plane
+
+You have to carry out the following steps on the master:
+
+1. Upgrade `kubectl` using [curl](/docs/tasks/tools/install-kubectl/#install-kubectl-binary-via-curl). Note: DO NOT use `apt` or `yum` or any other package manager to upgrade it.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: This is unnecessary. Only kubeadm is needed

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+2. Install the most recent version of `kubeadm` using curl.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Please tell us how to do that:
```console
$ export VERSION=v1.8.0 # or any given released k8s version
$ export ARCH=amd64 # or arm, arm64, ppc64le or s390x
$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+Also, note that only one minor version upgrade is supported. That is, you can only upgrade from, say 1.7 to 1.8, not from 1.7 to 1.9.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Upgrading your control plane
+
+You have to carry out the following steps on the master:
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: **by executing these commands on your master node**

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+This guide is for upgrading `kubeadm` clusters from version 1.7.x to 1.8.x.
+See also [upgrading kubeadm clusters from 1.6 to 1.7](/docs/tasks/administer-cluster/kubeadm-upgrade-1-7/) if you're on a 1.6 cluster currently.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a `kubeadm` Kubernetes cluster running version 1.7.0 or higher in order to use the process described here.
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: ...a functional `kubeadm` Kubernetes... ?

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+[upgrade/health] Checking Node health: All Nodes are healthy
+[upgrade/health] Checking if control plane is Static Pod-hosted or Self-Hosted: Static Pod-hosted.
+[upgrade/health] NOTE: kubeadm will upgrade your Static Pod-hosted control plane to a Self-Hosted one when upgrading if --feature-gates=SelfHosting=true is set (which is the default)
+[upgrade/health] If you strictly want to continue using a Static Pod-hosted control plane, set --feature-gates=SelfHosting=true when running 'kubeadm upgrade apply'
+[upgrade/health] Checking Static Pod manifests exists on disk: All required Static Pod manifests exist on disk
+[upgrade] Making sure the configuration is correct:
+[upgrade/config] Reading configuration from the cluster (you can get this with 'kubectl -n kube-system get cm kubeadm-config -oyaml')
+[upgrade] Fetching available versions to upgrade to:
+[upgrade/versions] Cluster version: v1.7.1
+[upgrade/versions] kubeadm version: v1.8.0-alpha.2.789+11f48dc291fe93
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Let's fake all these values to use v1.8.0 :)

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+	kubeadm upgrade apply --version v1.7.3
+```
+
+The `kubeadm upgrade plan` checks that your cluster is in an upgradeable state and fetches the versions available to upgrade to in an user-friendly way.
+
+4. Pick a version to upgrade to and run, for example, `kubeadm upgrade apply` as follows:
+
+```shell
+$ kubeadm upgrade apply --version v1.7.3
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Let's upgrade to v1.8.0 here in the docs

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-controller-manager
+[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
+[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backuped old manifest to "/tmp/kubeadm-upgrade830923296/old-manifests/kube-scheduler.yaml"
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-scheduler
+[apiclient] Found 0 Pods for label selector component=kube-scheduler
+[apiclient] Found 1 Pods for label selector component=kube-scheduler
+[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
+[self-hosted] Created TLS secret "ca" from ca.crt and ca.key
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Should not be a self-hosted cluster in the example

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+[self-hosted] self-hosted kube-controller-manager ready after 15.804586 seconds
+[apiclient] Found 0 Pods for label selector k8s-app=self-hosted-kube-scheduler
+[apiclient] Found 1 Pods for label selector k8s-app=self-hosted-kube-scheduler
+[self-hosted] self-hosted kube-scheduler ready after 18.958074 seconds
+[apiconfig] Created RBAC rules
+[addons] Applied essential addon: kube-proxy
+[addons] Applied essential addon: kube-dns
+```
+
+The `kubeadm upgrade apply` does the following:
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: remove `The`?

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+$ export VERSION=v1.8.0 # or any given released Kubernetes version
+$ export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm
+```
+
+1. On the master node, run the following:
+
+```shell
+$ kubeadm upgrade plan
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: First of all, you have to remember what CLI args you passed to `kubeadm init` the first time.
If you used flags, do this `kubeadm config upload from-flags [flags]`, flags can be empty
If you used a config file, do this `kubeadm config upload from-file --config [config]`, config mandatory

This is to preserve the config for future upgrades. Note that this has only to be done on the **first** time you use `kubeadm upgrade`

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+[upgrade] Making sure the cluster is healthy:
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Replace with 
```
[preflight] Running pre-flight checks
[upgrade] Making sure the cluster is healthy:
[upgrade/health] Checking API Server health: Healthy
[upgrade/health] Checking Node health: All Nodes are healthy
[upgrade/health] Checking Static Pod manifests exists on disk: All manifests exist on disk
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
[upgrade] Fetching available versions to upgrade to:
[upgrade/versions] Cluster version: v1.7.1
[upgrade/versions] kubeadm version: v1.8.0
[upgrade/versions] Latest stable version: v1.8.0
[upgrade/versions] Latest version in the v1.7 series: v1.7.6

Components that must be upgraded manually after you've upgraded the control plane with 'kubeadm upgrade apply':
COMPONENT   CURRENT      AVAILABLE
Kubelet     1 x v1.7.1   v1.7.6

Upgrade to the latest version in the v1.7 series:

COMPONENT            CURRENT   AVAILABLE
API Server           v1.7.1    v1.7.6
Controller Manager   v1.7.1    v1.7.6
Scheduler            v1.7.1    v1.7.6
Kube Proxy           v1.7.1    v1.7.6
Kube DNS             1.14.4    1.14.4

You can now apply the upgrade by executing the following command:

	kubeadm upgrade apply v1.7.6

_____________________________________________________________________

Components that must be upgraded manually after you've upgraded the control plane with 'kubeadm upgrade apply':
COMPONENT   CURRENT      AVAILABLE
Kubelet     1 x v1.7.1   v1.8.0

Upgrade to the latest experimental version:

COMPONENT            CURRENT   AVAILABLE
API Server           v1.7.1    v1.8.0
Controller Manager   v1.7.1    v1.8.0
Scheduler            v1.7.1    v1.8.0
Kube Proxy           v1.7.1    v1.8.0
Kube DNS             1.14.4    1.14.4

You can now apply the upgrade by executing the following command:

	kubeadm upgrade apply v1.8.0

Note: Before you do can perform this upgrade, you have to update kubeadm to v1.8.0

_____________________________________________________________________


```

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+
+	kubeadm upgrade apply --version v1.7.3
+```
+
+The `kubeadm upgrade plan` checks that your cluster is in an upgradeable state and fetches the versions available to upgrade to in an user-friendly way.
+
+1. Pick a version to upgrade to and run, for example, `kubeadm upgrade apply` as follows:
+
+```shell
+$ kubeadm upgrade apply --version v1.8.0
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: now only `kubeadm upgrade apply v1.8.0`

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+[upgrade] Making sure the cluster is healthy:
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: Change output to
```
[preflight] Running pre-flight checks
[upgrade] Making sure the cluster is healthy:
[upgrade/health] Checking API Server health: Healthy
[upgrade/health] Checking Node health: All Nodes are healthy
[upgrade/health] Checking Static Pod manifests exists on disk: All manifests exist on disk
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
[upgrade/version] You have chosen to upgrade to version "v1.8.0"
[upgrade/versions] Cluster version: v1.7.1
[upgrade/versions] kubeadm version: v1.8.0
[upgrade/prepull] Will prepull images for components [kube-apiserver kube-controller-manager kube-scheduler]
[upgrade/prepull] Prepulling image for component kube-scheduler.
[upgrade/prepull] Prepulling image for component kube-apiserver.
[upgrade/prepull] Prepulling image for component kube-controller-manager.
[apiclient] Found 0 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
[apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
[apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-apiserver
[apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-controller-manager
[upgrade/prepull] Prepulled image for component kube-apiserver.
[upgrade/prepull] Prepulled image for component kube-controller-manager.
[upgrade/prepull] Prepulled image for component kube-scheduler.
[upgrade/prepull] Successfully prepulled the images for all the control plane components
[upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.8.0"...
[upgrade/staticpods] Writing upgraded Static Pod manifests to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests432902769"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests432902769/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests432902769/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests432902769/kube-scheduler.yaml"
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-apiserver.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests155856668/kube-apiserver.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-apiserver
[upgrade/staticpods] Component "kube-apiserver" upgraded successfully!
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-controller-manager.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests155856668/kube-controller-manager.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-controller-manager
[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests155856668/kube-scheduler.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
[apiclient] Found 1 Pods for label selector component=kube-scheduler
[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
[addons] Applied essential addon: kube-dns
[addons] Applied essential addon: kube-proxy

[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.8.0". Enjoy!

[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
```

File: docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+$ kubectl cordon $WORKER
+$ kubectl drain $WORKER
+```
+
+1. Upgrade the `kubelet` version on the `$WORKER` node, either by using a Linux distribution-specific package manager such as `apt-get` or `yum` or manually as described in the following:
+
+```shell
+$ sudo systemctl stop kubelet
+$ curl -s -L -o kubelet \
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-cmd.md]: This won't work, as the kubelet droping manifest also gets updated.
At this stage, run the full `apt-get upgrade` or `yum update` and write that here.
You may use tabs for this, look at the main kubeadm doc how to do so

**Comment by mhausenblas** [comment]
@luxas I've addressed you latest round of comments now

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md [hunk 19]
```diff
@@ -0,0 +1,254 @@
...
+Controller Manager   v1.7.1    v1.8.0
+Scheduler            v1.7.1    v1.8.0
+Kube Proxy           v1.7.1    v1.8.0
+Kube DNS             1.14.4    1.14.4
+
+You can now apply the upgrade by executing the following command:
+
+	kubeadm upgrade apply v1.8.0
+
+Note: Before you do can perform this upgrade, you have to update kubeadm to v1.8.0
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: nit: this line should be removed, missed that when I posted this to you

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md [hunk 20]
```diff
@@ -0,0 +1,254 @@
...
+   Your Container Network Interface (CNI) provider might have its own upgrade instructions to follow now.
+   Check the [addons](/docs/concepts/cluster-administration/addons/) page to
+   find your CNI provider and see if there are additional upgrade steps
+   necessary.
+
+## Upgrading your worker nodes
+
+For each worker node (referred to as `$WORKER` below) in your cluster, upgrade `kubelet` by executing the following commands:
+
+1. Prepare the node for maintenance, marking it unschedulable and evicting the workload:
```
Comment (luxas) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: nit: this has to be done on the master

**Comment by luxas** [comment]
Merging this based on @chenopis docs review.
You can address the really small nits in a follow up if you want.

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md (hunk 20 above)
Comment (mhausenblas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: Hmmm, if I understand you right then it's probably more than a nit ;)

So, are you saying this has to be done *as well* on the master or *only* on the master. If the former, then no problem, if the latter then this shouldn't be in a separate section at all but in the previous section and all references to worker nodes removed. Please advise @luxas …

Comment (luxas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: I'm saying that this step _might_ be of use if you want to upgrade your master kubelet, yes.
But in particular, I referred to that these `kubectl` commands can only be done _on the master_
After you've run drain/cordon WORKER1 on the master, go ahead and do `apt-get install kubelet` on WORKER1. Finally, go back to the master and uncordon.

Makes sense?

File: docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md [hunk 21]
```diff
@@ -0,0 +1,254 @@
...
+
+```shell
+$ kubectl cordon $WORKER
+$ kubectl drain $WORKER
```
Comment (kargakis) [inline docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: I think `drain` is cordoning the node.

Comment (mhausenblas) [reply docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md]: Yeah @kargakis that was my hunch as well but since @luxas and I were not 100% sure about it we decided to go with the 'safe' option. Happy to change, though …
